pkg/corerp/api/v20231001preview/environment_conversion.go

```diff
@@ -180,6 +183,59 @@ func (dst *EnvironmentResource) ConvertFrom(src v1.DataModelInterface) error {
 	return nil
 }
 
+func toRecipeConfigDatamodel(config *RecipeConfigProperties) datamodel.RecipeConfigProperties {
+	if config != nil {
+		recipeConfig := datamodel.RecipeConfigProperties{}
+		if config.Terraform != nil {
+			recipeConfig.Terraform = datamodel.TerraformConfigProperties{}
+			if config.Terraform.Authentication != nil {
+				recipeConfig.Terraform.Authentication = datamodel.AuthConfig{}
+				gitConfig := config.Terraform.Authentication.Git
+				if gitConfig != nil {
+					recipeConfig.Terraform.Authentication.Git = datamodel.GitAuthConfig{}
+					if gitConfig.Pat != nil {
+						p := map[string]datamodel.Secret{}
+						for k, v := range gitConfig.Pat {
+							p[k] = datamodel.Secret{
+								SecretStore: to.String(v.SecretStore),
+							}
+						}
+						recipeConfig.Terraform.Authentication.Git.PAT = p
+					}
+
+				}
+			}
+		}
+		return recipeConfig
+	}
+	return datamodel.RecipeConfigProperties{}
+}
+
+func fromRecipeConfigDatamodel(config datamodel.RecipeConfigProperties) *RecipeConfigProperties {
+	if !reflect.DeepEqual(config, datamodel.RecipeConfigProperties{}) {
+		recipeConfig := &RecipeConfigProperties{}
+		if !reflect.DeepEqual(config.Terraform, datamodel.TerraformConfigProperties{}) {
+			recipeConfig.Terraform = &TerraformConfigProperties{}
+			if !reflect.DeepEqual(config.Terraform.Authentication, datamodel.AuthConfig{}) {
+				recipeConfig.Terraform.Authentication = &AuthConfig{}
+				if !reflect.DeepEqual(config.Terraform.Authentication.Git, datamodel.GitAuthConfig{}) {
+					recipeConfig.Terraform.Authentication.Git = &GitAuthConfig{}
+					if config.Terraform.Authentication.Git.PAT != nil {
+						recipeConfig.Terraform.Authentication.Git.Pat = map[string]*Secret{}
+					}
+				}
+			}
+		}
+		for k, v := range config.Terraform.Authentication.Git.PAT {
```

Comment:
Would this throw a nil pointer exception if config doesn't have Terraform map? Or Terraform map doesn't have Authentication and so on. We can add unit tests for those cases.

Reply:
I think this for loop should move to the **last if** in above. See: **if config.Terraform.Authentication.Git.PAT != nil**.